src/platform/middleware.ts

```diff
@@ -73,3 +75,16 @@ export const gracefullyDenyDeletedDevices: RequestHandler = (
 			next();
 		});
 };
+
+export const registerDeviceContact = (path: string) => {
+	return ((req, _res, next) => {
+		const uuid = _.get(req, path, '');
+		if (uuid !== '') {
+			DeviceOnlineState.getPollIntervalForDevice(uuid).then(pollInterval =>
+				DeviceOnlineState.manager.captureEventFor(uuid, pollInterval / 1000),
+			);
```

Comment:
Should probably have a `.catch`/report here rather than potentially an `UnhandledPromiseRejection`